### DISCUSSION TITLE
Homebrew install in Mac

### DISCUSSION THREAD

**Discussion by woshiyc123** [Homebrew/discussions#158] [Everyday usage]
Hi Guys
I was having problem when I was installing the homebrew on my Mac 
when I copy phase the link on the homebrew homepage, it shows me like that.

deMacBook-Pro:~ moumou$ curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh
curl: (7) Failed to connect to raw.githubusercontent.com port 443: Connection refused 

And I have tried to download install.sh by opening the webpage on my browser, and I run install.sh in my terminal, It works, but it is like 2k/s when downloading it.

Then I have tried to download from here(GitHub) , I have download the folder 'homebred-core-master', and I tried to do 'brew tap mongodb/brew'
 
it starting working but later it gives me error 

deMacBook-Pro:~ moumou$ brew tap mongodb/brew
Updating Homebrew...
==> Homebrew has enabled anonymous aggregate formula and cask analytics.
Read the analytics documentation (and how to opt-out) here:
  https://docs.brew.sh/Analytics
No analytics have been recorded yet (or will be during this `brew` run).

==> Tapping homebrew/core
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core'...
remote: Enumerating objects: 35, done.
remote: Counting objects: 100% (35/35), done.
remote: Compressing objects: 100% (25/25), done.
error: RPC failed; curl 18 transfer closed with outstanding read data remaining
fatal: the remote end hung up unexpectedly
fatal: early EOF
fatal: index-pack failed
Error: Failure while executing; `git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core` exited with 128.
Error: Failure while executing; `/usr/local/bin/brew tap homebrew/core` exited with 1.

Can someone help me ?
Thank you

**Comment by SMillerDev** [comment]
This is an issue with your internet connection, there's not much I can do to help you debug it. Try changing networks or temporarily disabling any firewalls.

**Reply by woshiyc123** [reply]
okay I will try thank you

**Reply by rhanneken** [reply]
I don't think it's an issue with @woshiyc123's Internet connection. I had the same problem last night. I think it's a rerun of [this issue](https://github.com/Homebrew/homebrew-core/issues/61772), which has apparently been a recurring problem on the GitHub side.

I worked around it by [adding my SSH public key to my GitHub account](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/connecting-to-github-with-ssh) and executing the following two commands: 

```sh
grep -qF 'github.com' ~/.ssh/known_hosts || ssh-keyscan github.com >> ~/.ssh/known_hosts
HOMEBREW_CORE_GIT_REMOTE=git@github.com:Homebrew/homebrew-core.git brew update --force
```

I suppose the exact `brew` command is unimportant. The important part is the part that sets the environment variable `HOMEBREW_CORE_GIT_REMOTE`. This changes the Git remote for the homebrew-core repository so that Homebrew interacts with GitHub using SSH instead of HTTP (just for homebrew-core).

If GitHub fixes this issue, I assume you can restore the default remote URL for homebrew-core by doing something like 

```sh
HOMEBREW_CORE_GIT_REMOTE=https://github.com/Homebrew/homebrew-core brew update
```

**Reply by woshiyc123** [reply]
I have solved this problem , I download the package from GitHub and then use command line ' git clone https://github.com/Homebrew/homebrew-core /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core --depth=1'
This works with download speed 10kb/s.
